snippets/deepFreeze.md

```diff
@@ -0,0 +1,38 @@
+### deepFreeze
+
+Deep freezes an object.
+
+Calls `Object.freeze(obj)` recursively on all unfrozen properties of obj that are `typeof` function or object.
+
+```js
+const deepFreeze = obj => {
+  Object.freeze(obj);
+
+  Object.getOwnPropertyNames(obj).forEach(function(prop) {
+    if (
+      obj.hasOwnProperty(prop) &&
+      obj[prop] !== null &&
```

Comment:
Provided we only want `obj[prop] instanceof Object` properties, do we need to check for `null`? I mean it's (as far as a I can tell) impossible for a property to be `null` and `instanceof Object` at the same time.

Reply:
You are correct in assuming that null is not an instance of Object

Reply:
@skatcat31 Check my proposal below, I think it makes the code a lot shorter and easier to understand.